pydantic_ai_slim/pydantic_ai/models/anthropic.py

```diff
@@ -409,13 +409,29 @@ def _map_usage(message: AnthropicMessage | RawMessageStreamEvent) -> usage.Usage
     if response_usage is None:
         return usage.Usage()
 
+    # Store all integer-typed usage values in the details dict
+    response_usage_dict = response_usage.model_dump()
+    details: dict[str, int] = {}
+    for key, value in response_usage_dict.items():
+        if isinstance(value, int):
+            details[key] = value
+
+    # Usage coming from the RawMessageDeltaEvent doesn't have input token data, hence these getattr calls
     request_tokens = getattr(response_usage, 'input_tokens', None)
```

Comment:
Can we simplify this by defaulting to `0` here and below, and skipping the `isinstance` calls?

Reply:
We can, and that will definitely be simpler, but it might change behavior elsewhere. I did it this way specifically since `usage.Usage.request_tokens` supports `None`, and so do the `cache_*` fields on the Anthropic `Usage` object and wanted to represent them faithfully. If I coerce `request_tokens` to `0`, the `usage.Usage` object will tell callers that there is a count of 0 tokens rather than that attribute being missing (such as when processing a `RawMessageDeltaEvent` event).

That said, more than happy to make this change, just want to make sure this was considered

Reply:
I believe this as simple as it can be, because `cache_creation_input_tokens` and `cache_read_input_tokens` may or may not exist on the `response_usage` and when they do exist, they can be either an integer or `None`. But this still has the downside of setting `Usage.request_tokens=0` when it's actually `None`

```python
    request_tokens = (
        getattr(response_usage, 'input_tokens', 0)
        + (getattr(response_usage, 'cache_creation_input_tokens', 0) or 0)
        + (getattr(response_usage, 'cache_read_input_tokens', 0) or 0)
    )

    return usage.Usage(
        request_tokens=request_tokens,
        response_tokens=response_usage.output_tokens,
        total_tokens=request_tokens + response_usage.output_tokens,
        details=details or None,
    )

Reply:
@DouweM implemented this ^ change

Reply:
@timesler What do you think of changing `0` back into `None` in the arguments to `Usage`? 0 is basically never going to be accurate.

Reply:
I'm not sure we can assume that, but we can definitely recheck if the attribute is defined on the input object and set it to None if not